### PR TITLE
Preserve tilt angles from utils.set_tilt_angles

### PR DESCRIPTION
**Pull request by mathturtle**
@yijiang1 For #442, a (hopefully) minor fix.  Please test everything you can think of to do with tilt angles to be sure we don't miss anything else.  I will be doing the same.

### CODE REVIEW

**Comment by yijiang1** [comment]
This fixes #442 but reveals another bug probably related to #445 .
One consistent way to reproduce the bug:
1. Load a dataset.
2. Click "Generate Tilt Series" under "Tomography" menu.
When finished, the dataset is marked as a tilt series and tilt angles are updated like this
screen shot 2016-06-23 at 3 22 23 pm

4. Now, click "Reset" under "File" menu.
5. Repeat step 1 and 2.
When finished, the tilt angles are updated but the dataset is NOT marked as tilt series.
screen shot 2016-06-23 at 3 23 13 pm

**Comment by mathturtle** [comment]
Updated branch.  Here is a slightly cleaner way to fix it.  It doesn't help the #445-related bug you found though.  That appears to be an update issue where the menu items don't get the dataset-was-updated signal correctly.  The data source is marked as a tilt series (otherwise the tilt angles wouldn't show up on the properties panel), but the menu wasn't notified.

File: tomviz/DataSource.cxx [hunk 1]
```diff
@@ -66,13 +66,19 @@ class DataSource::DSInternals
     vtkFieldData* fd = data->GetFieldData();
     if (!this->TiltAngles)
     {
+      if (!fd->HasArray("tilt_angles"))
```
Comment (cryos) [inline tomviz/DataSource.cxx]: This does not seem to do anything.

Comment (mathturtle) [inline tomviz/DataSource.cxx]: I've deleted it, and updated the branch.  I think this was left over from the first implementation of this branch.  I copied the contents out into the containing block and I guess I forgot to delete this.

**Pull request merged**